core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java

```diff
@@ -85,10 +85,14 @@ public FileSystemPermission getPermission() {
    * Applies umask.
    *
    * @param umask the umask to apply
+   * @param configuration the configuration
    * @return a new {@link PermissionStatus}
    * @see FileSystemPermission#applyUMask(FileSystemPermission)
    */
-  public PermissionStatus applyUMask(FileSystemPermission umask) {
+  public PermissionStatus applyUMask(FileSystemPermission umask, Configuration configuration) {
+    if (!SecurityUtils.isAuthorizationEnabled(configuration)) {
```

Comment:
I feel this check should not be here inside `applyUMask` but on the caller side.
The reason is that, by having this, when security is not turned on, `PermissionStatus.applyUMask` will become no-op, which makes less sense. However, I feel this `applyUMask` logic should still hold as it is a part of `PermissionStatus` logic, it should not depend on whether security is turned on or off.